nat/traversal/pinger.go

```diff
@@ -176,7 +178,7 @@ func (p *Pinger) PingProvider(ip string, port int) error {
 	if err != nil {
 		return errors.Wrap(err, "failed to get connection")
 	}
-	defer conn.Close()
+	//defer conn.Close()
```

Comment:
?
Maybe resource cleaning should happen as defer of routine `consumerHandOff()`

Reply:
yes, its happens inside `consumerHandOff(conn)` since closure may happen only on error or user's intent.